graal-js/src/com.oracle.js.parser/src/com/oracle/js/parser/Parser.java

```diff
@@ -6329,7 +6359,12 @@ private FunctionNode module(final String moduleName) {
 
             addFunctionDeclarations(script);
         } finally {
-            functionDeclarations = null;
+            if (moduleBlock) {
+                functionDeclarations = Collections.emptyList();
+            } else {
+                functionDeclarations = null;
+            }
```

Comment:
couldn't we always use `null` and create the empty collection for module blocks later on where needed? Otherwise you can consider using `functionDeclarations = moduleBlock ? Collections.emptyMap() : null;` and add a one-line comment to signal that module blocks expect an empty collection

Reply:
Null will result in a cascade of Errors. Used your suggestion with an emptyList.